### ISSUE TITLE
UnicodeEncodeError while building in a directory with a path containing non-ascii characters

### ISSUE THREAD

**Issue by maxatwork** [docker/compose#2077]
Trying to perform `docker-compose build` in a directory with a path containing non-ascii characters (Russian letters), got this:

```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 39, in main
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 135, in build
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.project", line 233, in build
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.service", line 701, in build
UnicodeEncodeError: 'ascii' codec can't encode characters in position 17-22: ordinal not in range(128)
```

Doing the same in a directory without non-ascii characters in a path seems to be OK. 

Mac OS X 10.10.5
Python 2.7.10
docker 1.8.1, build d12ea79
docker-machine 0.4.1 (e2c88d6)
docker-compose v1.4.0

**Comment by awolfe-silversky** [comment]
Got caught also.  Using greek mathematical symbols in path.  OS X 10.10

**Comment by dnephin** [comment]
I thought we had fixed this in the 1.5.0 or 1.5.1 release.  What version are you using?

**Comment by awolfe-silversky** [comment]
awolfe_7069: docker-compose --version
docker-compose version: 1.4.0

Comes from depending on mac brew I suppose...

On Nov 19, 2015, at 12:05, Daniel Nephin <notifications@github.com<mailto:notifications@github.com>> wrote:


I thought we had fixed this in the 1.5.0 or 1.5.1 release. What version are you using?

—
Reply to this email directly or view it on GitHub<https://github.com/docker/compose/issues/2077#issuecomment-158120278>.


Please consider the environment before printing this email. This message should be regarded as confidential. If you have received this email in error, please notify the sender and destroy it immediately. Statements of intent shall only become binding when confirmed in hard copy by an authorized signatory. The contents of this email may relate to dealings with other companies under the control of BAE Systems Applied Intelligence Limited, details of which can be found at http://www.baesystems.com/Businesses/index.htm.

**Comment by dnephin** [comment]
Confirmed this is fixed in 1.5.1, so going to close.

The recommended install option for OSX is https://www.docker.com/docker-toolbox but we also have these install options available: http://docs.docker.com/compose/install/
